src/OpenTelemetry.Instrumentation.ConfluentKafka/InstrumentedProducer.cs

```diff
@@ -0,0 +1,243 @@
+        }
+        finally
+        {
+            activity?.Stop();
+        }
+    }
+
+    public int Poll(TimeSpan timeout)
+    {
+        return producer.Poll(timeout);
+    }
+
+    public int Flush(TimeSpan timeout)
+    {
+        return producer.Flush(timeout);
+    }
+
+    public void Flush(CancellationToken cancellationToken = default)
+    {
+        producer.Flush(cancellationToken);
+    }
+
+    public void InitTransactions(TimeSpan timeout)
+    {
+        producer.InitTransactions(timeout);
+    }
+
+    public void BeginTransaction()
+    {
+        producer.BeginTransaction();
+    }
+
+    public void CommitTransaction(TimeSpan timeout)
+    {
+        producer.CommitTransaction(timeout);
+    }
+
+    public void CommitTransaction()
+    {
+        producer.CommitTransaction();
+    }
+
+    public void AbortTransaction(TimeSpan timeout)
+    {
+        producer.AbortTransaction(timeout);
+    }
+
+    public void AbortTransaction()
+    {
+        producer.AbortTransaction();
+    }
+
+    public void SendOffsetsToTransaction(IEnumerable<TopicPartitionOffset> offsets, IConsumerGroupMetadata groupMetadata, TimeSpan timeout)
+    {
+        producer.SendOffsetsToTransaction(offsets, groupMetadata, timeout);
+    }
+
+    public void Dispose()
+    {
+        producer.Dispose();
+    }
+
+    private Activity? StartActivity(string operation, string topic, Message<TKey, TValue> message)
+    {
+        var activity = ConfluentKafkaCommon.ActivitySource.StartActivity(string.Concat(topic, " ", operation), ActivityKind.Producer);
+        if (activity == null)
+        {
+            return null;
+        }
+
+        activity.SetTag(SemanticConventions.AttributeMessagingSystem, MessagingSystem);
+        activity.SetTag(SemanticConventions.AttributeMessagingClientId, this.Name);
+        activity.SetTag(SemanticConventions.AttributeMessagingDestinationName, topic);
+        activity.SetTag(SemanticConventions.AttributeMessagingOperation, operation);
+
+        if (message.Key != null)
+        {
+            activity.SetTag(SemanticConventions.AttributeMessagingKafkaMessageKey, message.Key);
+        }
+
+        return activity;
+    }
+
+    private Activity? StartActivity(string operation, TopicPartition topicPartition, Message<TKey, TValue> message)
+    {
+        var activity = this.StartActivity(operation, topicPartition.Topic, message);
+        if (activity == null)
+        {
+            return null;
+        }
+
+        activity.SetTag(SemanticConventions.AttributeMessagingKafkaDestinationPartition, topicPartition.Partition.Value);
+        return activity;
+    }
+
+    private void InjectActivity(Activity? activity, Message<TKey, TValue> message)
+    {
+        (ActivityContext activityContext, Baggage baggage) = activity != null
+            ? (activity.Context, Baggage.Current.SetBaggage(activity.Baggage))
+            : (default, default);
```

Comment:
```suggestion
        (ActivityContext activityContext, Baggage baggage) = activity != null
            ? (activity.Context, Baggage.Current)
            : (default, default);
```

This seems to be trying to stuff `Activity.Baggage` into OTel `Baggage`. That _probably_ is a bad idea because AFAIK no other instrumentation does that. We generally recommend users do not use `Activity.Baggage` at all.

/cc @cijothomas